huaweicloud/services/dis/resource_huaweicloud_dis_stream.go

```diff
@@ -233,6 +258,7 @@ func resourceDisStreamRead(ctx context.Context, d *schema.ResourceData, meta int
 		d.Set("partition_count", detail.WritablePartitionCount),
 		d.Set("status", detail.Status),
 		d.Set("stream_id", detail.StreamId),
+		queryAndSetPartitionsToState(client, d, detail.StreamName),
```

Comment:
seems you ca set `partitions` directly here.

Reply:
if HasMorePartitions =true ,need to query next page